packages/framework/aqueduct/src/legacy-container-runtime-factories/legacyBaseContainerRuntimeFactory.ts

```diff
@@ -0,0 +1,109 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IContainerContext, IRuntime, IRuntimeFactory } from "@fluidframework/container-definitions";
+import {
+    IContainerRuntimeOptions,
+    FluidDataStoreRegistry,
+    makeLegacyContainerRuntime,
+} from "@fluidframework/container-runtime";
+import {
+    IContainerRuntime,
+} from "@fluidframework/container-runtime-definitions";
+import {
+    RuntimeRequestHandler,
+    buildRuntimeRequestHandler,
+    innerRequestHandler,
+} from "@fluidframework/request-handler";
+import {
+    IFluidDataStoreRegistry,
+    IProvideFluidDataStoreRegistry,
+    NamedFluidDataStoreRegistryEntries,
+} from "@fluidframework/runtime-definitions";
+import { DependencyContainer, DependencyContainerRegistry } from "@fluidframework/synthesize";
+
+/**
+ * BaseContainerRuntimeFactory produces container runtimes with a given data store and service registry, as well as
+ * given request handlers.  It can be subclassed to implement a first-time initialization procedure for the containers
+ * it creates.
+ * @deprecated Only use if your scenario requires backwards compatibility with documents that were produced before
```

Comment:
can you please double check that deprecated is picked up by vs code here? I.e. should be also on constructor mayb?

Reply:
It kind of does -- gets the strikethrough, but looks like the deprecation message only appears on non-constructor references to the class.  I'll add it to the constructor as well.